test/inductor/test_torchinductor.py

```diff
@@ -7576,6 +7578,358 @@ def test_rnn_compile_safe(self):
             x = torch.rand(1024, 20, 16).to(device)
             model(x)
 
+    def get_all_cudagraph_segments():
+        segments = torch.cuda.memory_snapshot()
+        return [segment for segment in segments if segment["segment_pool_id"] != (0, 0)]
+
+    def all_live_blocks():
+        blocks_addrs = []
+        for segment in get_all_cudagraph_segments():
+            addr = segment["address"]
+            for block in segment["blocks"]:
+                if block["state"] == "active_allocated":
+                    blocks_addrs.append(addr)
+                addr += block["size"]
+
+        return blocks_addrs
+
+    def all_live_block_count():
+        return len(all_live_blocks())
+
+    class CudaGraphTapeTests(TestCase):
```

Comment:
Let's create a new file for these tests.  `test_torchinductor.py` is already too big.

Reply:
Will do as another pr in the stack, and land all at once, to minimize churn to other prs

Reply:
I think you should squash all the PRs in the stack